src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -977,10 +983,12 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         //          > val vparamSymssOrEmptyParamsFromOverride =
         //          This means an accessor that overrides a Java-defined method gets a MethodType instead of a NullaryMethodType, which breaks lots of assumptions about accessors)
         def checkCanAutoApply(): Boolean = {
-          if (sourceLevel2_14 && !meth.isJavaDefined)
+          val afterExtMethods = phaseId(currentPeriod) >= currentRun.phaseNamed("extmethods").id
+          if (sourceLevel2_14 && !afterExtMethods && !matchNullaryLoosely) {
```

Comment:
I should have suggested `!isPastTyper`, it seems that's what's used for guarding warnings. Will push a fix.

Reply:
> I should have suggested `!isPastTyper`, it seems that's what's used for guarding warnings.

Side: I wonder if that (1) can and (2) should be always considered in the behaviour of `context.deprecationWarning`.

Reply:
Uncurry has one. Deprecation is used for all sorts of behaviors, and not just API. I'm not sure that is kosher, but I'm not a purist.